simulator/func_sim/alu.h

```diff
@@ -240,6 +240,18 @@ struct ALU
     // Bit manipulations
     template<typename I> static void sbext( I* instr) { instr->v_dst = 1 & ( instr->v_src1 >> shamt_v_src2<typename I::RegisterUInt>( instr)); }
 
+    template<typename I, typename T>
+    static void clmul( I* instr)
+    {
+        instr->v_dst = 0;
+        std::size_t max = bitwidth<T> - count_leading_zeroes<T>( instr->v_src2);
```

Comment:
Could you please share the motivation for that adjustment?

Reply:
It's optimisation for numbers with a lot of leading zeros. Without it loop do nothing at the end of numbers like 0x0000'000A.

Reply:
There are a lot of objections:
1. Code becomes a little bit more fragile and less obvious
2. Branch prediction for loop exit might operate poorly 
4. Amdahl's law: optimization of rare instruction `clmul` won't significantly boost up simulation
3. On default code, RISC-V-targeted compiler might detect pattern and replace code with a single `clmul`. With optimization, chances are much less.

Usually, the first question of optimization is quantitative estimation of speed up. If optimization is proved to be significant, then we can throw away the objections.
I suggest to keep code more straightforward.